src/firmware_variables/variables.py

```diff
@@ -86,3 +89,65 @@ def delete_variable(name, *args, **kwargs):
     :param attributes: @see Attributes
     """
     set_variable(name, value=b"", *args, **kwargs)
+
+
+class NTError(OSError):
+    def __init__(self, nt_status):
+        self.nt_status = nt_status
+
+    def __str__(self):
+        return "[Error 0x{:08X}]".format(self.nt_status)
+
+
+def get_all_variables_names():
+    """
+    Get the names of all the UEFI Variables in the system.
+    :return: A list of tuples containing namespace and variable name.
+    """
+    INFORMATION_VARIABLE_NAMES = 1
+    STATUS_SUCCESS = 0
+    STATUS_BUFFER_TOO_SMALL = 0xc0000023
+    SIZEOF_GUID = 16
+    SIZEOF_DWORD = 4
+
+    def parse_variable_entry(entry_data):
```

Comment:
why inline functions? move out and add "_" prefix or move to different file

Reply:
because they are only relevant to this function, so there is no reason move them out